grievance_social_protection/rights.py

```
@@ -0,0 +1,207 @@
+            if permissions:
+                # Convert back to list format if still dict
+                if isinstance(permissions, dict):
+                    permissions = list(permissions.keys())
+                
+                category_info['generated_rights'] = {}
+                safe_name = category_name.lower().replace(' ', '_').replace('|', '_')
+                
+                for perm_type in permissions:
+                    # Generate codename
+                    codename = f"{perm_type}_{safe_name}_grievance"
+                    
+                    # Check if permission already exists
+                    if codename in existing_by_codename:
+                        perm = existing_by_codename[codename]
+                        category_info['generated_rights'][perm_type] = perm.id
+                        logger.info(f"Using existing permission: {codename} (ID: {perm.id})")
+                    else:
+                        # Generate new ID following suffix pattern
+                        try:
+                            new_id = cls._get_next_available_id(perm_type, used_ids)
+                            
+                            # Create new permission
+                            perm = Permission.objects.create(
+                                id=new_id,
+                                codename=codename,
+                                name=f"Can {perm_type.replace('_', ' ')} {category_name} tickets",
+                                content_type=ct
+                            )
+                            
+                            category_info['generated_rights'][perm_type] = perm.id
+                            used_ids.add(new_id)
+                            logger.info(f"Created new permission: {codename} (ID: {new_id})")
+                        except ValueError as e:
+                            logger.error(f"Failed to create permission {codename}: {e}")
+                            continue
```

When a permission failed to be created due to IDs are exhausted should we consider not catch the error and just let it throw there? Additionally, should we consider wrapping all Permission generations in the for loop in a transaction so either they all succeed or fail together? This way if one or a few were to fail we just clean up and redo the whole batch?